src/db/schema.prisma

```diff
@@ -63,3 +65,118 @@ model RefreshToken {
+model JobRequest {
+    id            Int      @id @default(autoincrement())
+    created_at    DateTime @default(now())
+    // Who made the request
+    user          User     @relation(fields: [user_id], references: [id])
+    user_id       Int
+    // What type of job was requested
+    type          JobType
+    // The input parameters requested
+    input_payload Json
+    // Did this request hit an existing cache entry?
+    cache_hit     Boolean
+    // If cache hit, what was the existing job?
+    // If cache miss, what new job was created?
+    job           Job      @relation(fields: [job_id], references: [id])
+    job_id        Int
+}
+
+model Job {
+    id            Int             @id @default(autoincrement())
+    created_at    DateTime        @default(now())
+    updated_at    DateTime        @updatedAt
+    type          JobType
+    status        JobStatus       @default(PENDING)
+    // What is the unique cache hash for this job?
+    hash          String
+    // Who created this job
+    user          User            @relation(fields: [user_id], references: [id])
+    user_id       Int
+    // Input parameters for the job - untyped JSON - application code manages
+    // zod IO schemas
+    input_payload Json
+    // Related entities
+    assignments   JobAssignment[]
+    results       JobResult[]
+
+    // Link to the original request that created this job
+    requests JobRequest[]
+}
+
+// JobAssignment tracks attempts to process a job
+model JobAssignment {
+    id              Int           @id @default(autoincrement())
+    created_at      DateTime      @default(now())
+    updated_at      DateTime      @updatedAt
+    // Link to the job
+    job             Job           @relation(fields: [job_id], references: [id])
+    job_id          Int
+    // ECS task info
+    ecs_task_arn    String
+    ecs_cluster_arn String
+    // When this assignment expires
+    expires_at      DateTime
+    // Where the worker should write results
+    storage_scheme  StorageScheme
+    storage_uri     String
+    // Assignment status tracking
+    heartbeat_at    DateTime?
+    completed_at    DateTime?
+    // Results from this assignment
+    result          JobResult?
+}
+
+// JobResult stores the output of a successful job execution
+model JobResult {
+    id             Int           @id @default(autoincrement())
+    created_at     DateTime      @default(now())
+    // Links to parent entities
+    job            Job           @relation(fields: [job_id], references: [id])
+    job_id         Int
+    // Which assignment produced this result?
+    assignment     JobAssignment @relation(fields: [assignment_id], references: [id])
+    assignment_id  Int           @unique
+    // Result data - optional
+    result_payload Json?
+    // Where results were stored
+    storage_scheme StorageScheme
+    storage_uri    String
+    // Optional metadata about the results
+    metadata       Json?
```

Comment:
I'm not really sure - it depends on how we want to use the software version - maybe it would go on the assignment (i.e. this worker is running @ a particular version), but could also be on the job type, or the result. It should probably be included in the hash for cache purposes, so it would need to be known about prior to any assignment...